src/CommandingOfficers/AW1/BM/Grit.java

```diff
@@ -66,7 +66,7 @@ public void modifyUnitAttack(StrikeParams params)
   public void modifyAttackRange(UnitContext uc)
   {
     if( uc.weapon != null && uc.weapon.rangeMax > 1 )
-      uc.weapon.rangeMax += 1;
+      uc.rangeMax += 1;
```

Comment:
I guess that's whatTwo things:
1) `WeaponModel` is mutable? I probably knew this at one point, but I also know I would only have named the class `*Model` if I intended for all of its members to be unchangeable.
2) Again, we probably hashed this out before, but why copy fields from `WeaponModel` into `UnitContext`, when the latter already owns a copy of the former? Just asking for trouble like this (especially when the immutable backstop isn't there).

Reply:
> `WeaponModel` is mutable?

The old design actually did modify the `WeaponModel`s and `MoveType`s directly for powers/D2Ds, which is why they had to be cloned for each CO's `UnitModel`s. I believe they're mostly still cloned because I'm not massively worried about a few bytes of duplicated data here or there.

Pretty much everything is mutable. We just don't change them during gameplay, because that would be weird.
I believe that's what we agreed on.
> I also know I would only have named the class *Model if I intended for all of its members to be unchangeable.

We _could_ make everything in the `*Model` classes immutable, but then we'd need some kind of builder pattern for Tech's mechs. I would be content with doing that via a library, but I really don't feel like writing that code myself.

> Again, we probably hashed this out before, but why copy fields from `WeaponModel` into `UnitContext`, when the latter already owns a copy of the former?

I imagine I did it to make it easy to substitute a `UnitContext` easily for either a `Unit` or a `UnitModel`.
I don't think we've hashed this one out, actually.
> Just asking for trouble like this (especially when the immutable backstop isn't there).

Fair criticism.